RxCocoa/iOS/UIBarButtonItem+Rx.swift

```diff
@@ -38,22 +38,11 @@ extension UIBarButtonItem {
     Reactive wrapper for target action pattern on `self`.
     */
     public var rx_tap: ControlEvent<Void> {
-        let source: Observable<Void> = Observable.create { [weak self] observer in
-
-            guard let control = self else {
-                observer.on(.Completed)
-                return NopDisposable.instance
-            }
-
-            let target = BarButtonItemTarget(barButtonItem: control) {
-                observer.on(.Next())
-            }
-            return target
-        }.takeUntil(rx_deallocated)
-        
-        return ControlEvent(events: source)
+        if let target = self.target as? BarButtonItemTarget {
+            return target.event
+        }
+        return BarButtonItemTarget(barButtonItem: self).event
```

Comment:
Thanks for your review, and great solution!
Of course, I make all of those changes, and commit this pr :3
Please wait...